### ISSUE TITLE
Support new format specification

### ISSUE THREAD

**Issue by repeatedly** [msgpack/msgpack-d#29]
See: https://github.com/msgpack/msgpack/blob/master/spec.md

Maybe, separating versions is needed for DUB.
0.9.x is for old specification.
1.x.x is for new specification.

**Comment by jwhear** [comment]
Do you have timeline for adding support for the new spec?  I'm looking at playing with Neovim's msgpack-rpc interface, but it requires spec v5 support.

**Comment by repeatedly** [comment]
Ah, I see.
I delayed the v5 spec support until someone want this because v4 spec is enough in D.
Okay, I will try implement it in this weekend.

**Comment by jwhear** [comment]
I tried parsing some data from Neovim and got segfaults, so I reckon it's due to it using the newer spec.  Are segfaults accepted behavior for spec-mismatch, or would you like me to open an issue with a reproduction case?

**Comment by repeatedly** [comment]
> would you like me to open an issue with a reproduction case?

Yes please. segfaults is unexpected behaviour.
Attaching actual binary data is awesome.

**Comment by jwhear** [comment]
Opened #45

**Comment by jwhear** [comment]
I am I right in assuming that msgpack-d doesn't currently support the Ext type family?  I'm using msgpack-rpc with an API and it works fine for many calls, but when the return value includes Ext objects, it gets stuck polling the underlying socket.

**Comment by repeatedly** [comment]
@jwhear Yes. Currently, msgpack-d doesn't support Ext type because there is no Ext type users with msgpack-d. Hmm... time to support v5 spec.
